### PR TITLE
Add indicators for visibility on project settings page

### PR DESCRIPTION
**Pull request by mineblock11**
## Information

Replaces the following text:

> Set the visibility of your project. Listed and archived projects are visible in search. 
Unlisted projects are published, but not visible in search or on user profiles. 
Private projects are only accessible by members of the project.

With some indicators that are much easier to understand.

Potentially could word differently? Not sure if the following should be used instead for the search and member profile bulletpoint

- Visible in search results
- Visible on member profiles

### CODE REVIEW

**Comment by Prospector** [comment]
I don't like how this doesn't tell you what each does until you click on it. You would have to select Unlisted to know what it did, for example. That's not intuitive because you'd want to know what it did before you selected it

**Comment by mineblock11** [comment]
Hm. Doesn't the user need to confirm their selection anyways via the "Save changes" button? Currently it's safe to switch between dropdown options without causing the visibility of the project to update.

**Comment by mineblock11** [comment]
I forgot to upload a video of it as well:

**Comment by Prospector** [comment]
Yeah but that's not intuitive to have to switch between options in order to figure out what they do. I think the display you added is fine, I'd just leave the old text (or an improvement upon it) as well.

**Comment by mineblock11** [comment]
I've updated it with a better description.

File: pages/[type]/[id]/settings/index.vue [hunk 1]
```diff
@@ -218,6 +253,9 @@ export default defineNuxtComponent({
     UploadIcon,
     SaveIcon,
     TrashIcon,
+    CrossIcon,
+    TickIcon,
```
Comment (Prospector) [inline pages/[type]/[id]/settings/index.vue:257]: I would rename this to CheckIcon to be more consistent but otherwise LGTM

**Review by Prospector — APPROVED** [review]

**Comment by Prospector** [comment]
Wait, why is exit.svg even a thing? We use `x.svg` for everything it seems.

**Comment by Minenash** [comment]
A problem with using a checklist looking system, besides what pro has mentioned, as that it could give the user a false impression of the actual state of the actual visibility (instead of the target visibility).

A possible idea would be to do something similar to the theme selector. Where each entry has information about it. Like it being listed in search/profile, who can see it, any warnings that the status displays

**Comment by mineblock11** [comment]
The actual visibility of the project is always listed underneath its name isn't it? I assume with the description above the checklist it would make it obvious that it's the target visibility.

A fix could be renaming it from Visibility -> Target Visibility

**Review by Prospector — COMMENTED** [review]
removing my approval

**Review by Prospector — CHANGES_REQUESTED** [review]
why can't you just remove an approval

**Comment by mineblock11** [comment]
Should be all good now

File: pages/[type]/[id]/settings/index.vue [hunk 2]
```diff
@@ -143,11 +143,43 @@
       </template>
       <div class="adjacent-input">
         <label for="project-visibility">
-          <span class="label__title">Visibility</span>
+          <span class="label__title">Target Visibility</span>
```
Comment (triphora) [inline pages/[type]/[id]/settings/index.vue:146]:
```suggestion
          <span class="label__title">Target visibility</span>
```

**Review by triphora — COMMENTED** [review]
Might not be worth fixing but there is a ton of unnecessary padding here
image

**Comment by Prospector** [comment]
I don't like the heading "Target visibility". Maybe instead, when your selected visibility is different from the current saved visibility, it changes the text from "Visible in search" to "Will be visible in search" to show that it's unsaved.

**Review by Prospector — CHANGES_REQUESTED** [review]
I don't like the heading "Target visibility". Maybe instead, when your selected visibility is different from the current saved visibility, it changes the text from "Visible in search" to "Will be visible in search" to show that it's unsaved.

**Comment by Prospector** [comment]
I think that's pretty good!

**Comment by Prospector** [comment]
Just rename back to Visibility

**Comment by brawaru** [comment]
I personally don't like text changing like this, but also when thinking about solution I can only remember a couple UIs that show a table (current / new), others instead use a popup that explains consequences (e.g. GitHub), since changing visibility is a bit dangerous action.

Screenshot of table layout concept showing two cards below the description for visibility option, where first card shows current visibility (not visible in search, nor via profile, and can only be accessed by members through URL) and new visibility (visible in search and on profile, and accessible through URL by everyone. In URL visibility who can view URL is marked with bold text. On the right arrows and square is drown above tho selector, suggesting that it should be moved slightly above.

**Comment by mineblock11** [comment]
This wouldn't fit on mobile, and I think its quite excessive for something so simple

**Review by Geometrically — APPROVED** [review]

**Pull request merged**